tests/unit/snow/test_sql_transpiler.py

```diff
@@ -89,3 +88,32 @@ def test_parse_sql_content(transpiler):
     result = list(transpiler.parse_sql_content("SELECT * FROM table_name", "test.sql"))
     assert result[0][0] == "table_name"
     assert result[0][1] == "test.sql"
+
+
+def test_safe_parse(transpiler, write_dialect):
+    dialect = transpiler.read_dialect
+    result, error = transpiler.safe_parse("SELECT col1 from tab1;SELECT11 col1 from tab2", dialect)
+    expected_result = [expressions.Column(this=expressions.Identifier(this="col1", quoted=False))]
+    expected_from_result = expressions.From(
+        this=expressions.Table(this=expressions.Identifier(this="tab1", quoted=False))
+    )
+    for exp in result:
+        if exp.parsed_expression:
+            print("yes")
+            assert repr(exp.parsed_expression.args["expressions"]) == repr(expected_result)
+            assert repr(exp.parsed_expression.args["from"]) == repr(expected_from_result)
+    assert "PARSING ERROR" in error[0]
+
+
+def test_safe_parse_with_semicolon(transpiler, write_dialect):
+    dialect = transpiler.read_dialect
+    result, error = transpiler.safe_parse("SELECT 'col1;col2' from tab1", dialect)
```

Comment:
nit: to make this much more understandable.

```suggestion
    result, error = transpiler.safe_parse("SELECT split(col2,';') from tab1 where col1 like ';%') ", dialect)
```

Reply:
done